Framework/PythonInterface/plugins/algorithms/LoadVesuvio.py

```diff
@@ -142,13 +142,46 @@ def PyExec(self):
     def validateInputs(self):
         issues = {}
 
-        # Validtae run number ranges
+        # Validate run number ranges
         run_str = self.getProperty(RUN_PROP).value
         if "-" in run_str:
             lower, upper = run_str.split("-")
             if upper < lower:
                 issues[RUN_PROP] = "Range must be in format lower-upper"
 
+
+        # Validate SpectrumList
+        speclists_str = self.getProperty(SPECTRA_PROP).value
+
+        if ";" in speclists_str:
+            # Split if in 2-3;6-7 format
+            speclists_str = speclists_str.split(";")
+        else:
+            # Treat spec as a list
+            speclists_str = [speclists_str]
+
+        for speclist_str in speclists_str:
+            if "-" in speclist_str:
+                lower, upper = speclist_str.split("-")
+                lower = int(lower)
+                upper = int(upper)
+                # Check format
+                if upper < lower:
+                    issues[SPECTRA_PROP] = "Range must be in format lower-upper"
+                # Check Min/Max boundaries
+                if lower < 3:
```

Comment:
In which case do you think it might be worth putting the functionality from `_raise_error_if_non_valid_mode` and `_raise_error_if_mix_fwd_back` into the validation function? That way it'll stop users from entering invalid inputs rather than throwing a RuntimeError in the alg.